doc/requirements/sequential_cutoff

```diff
@@ -0,0 +1,234 @@
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff operation
+id: operation
+---
+
+When lenght of the stream reaches the cutoff threshold, each next IO request
+which continues the stream shall be handled in Pass-Through mode.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff for reads and writes
+id: reads_and_writes
+---
+
+The sequential cutoff shall track streams of read and write IO requests
+separately.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Multiple per-CPU sequential cutoff streams
+id: per_cpu_multiple_streams
+---
+
+The sequential cutoff shall be able to detect up to 128 per-CPU streams.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Multiple shared sequential cutoff streams
+id: per_cpu_multiple_streams
+---
+
+The sequential cutoff shall be able to detect up to 64 shared streams.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff per-CPU stream continuation
+id: per_cpu_stream_continuation
+---
+
+When new IO request is being handled, the sequential cutoff shall first try to
+account it to one of per-CPU streams local to the CPU on which the IO request
+is being handled.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff shared stream continuation
+id: shared_stream_continuation
+---
+
+When new IO request is being handled and there is no per-CPU stream which can
+be continued by this IO request, the sequential cutoff shall try to account it
+to one of shared streams.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff stream creation
+id: stream_creation
+---
+
+When new IO request is being handled and there is neither per-CPU not shared
+stream which can be countinued by this IO request, the sequential cutoff shall
+create new per-CPU stream starting with this IO request.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Sequential cutoff stream promotion
+id: stream_promotion
+---
+
+The per-CPU stream shall be local to given CPU until number of IO requests
```

Comment:
You could split it to two state-driven requirements.

Reply:
Actually it turns out, that one even driven is enough.